airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py

```diff
@@ -24,10 +24,10 @@ class CheckStream(ConnectionChecker):
     stream_names: List[str]
     parameters: InitVar[Mapping[str, Any]]
 
-    def __post_init__(self, parameters: Mapping[str, Any]):
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
         self._parameters = parameters
 
-    def check_connection(self, source: Source, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+    def check_connection(self, source: DeclarativeSource, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
```

Comment:
`Source` seems preferable since this method doesn't depend on anything specific to `DeclarativeSource`.